rules/react.js

```diff
@@ -1,91 +1,146 @@
-		// Don't allow deprecated React practices
-		"react/no-deprecated": [2, {
-			"react": "0.14.0"
-		}],
+		"react/no-danger": "warn",
+		// Don't allow deprecated React methods.
+		"react/no-deprecated": "error",
 		// Prevent setState in componentDidMount
-		"react/no-did-mount-set-state": [2, "allow-in-func"],
+		"react/no-did-mount-set-state": "error",
 		// Prevent setState in componentDidUpdate
-		"react/no-did-update-set-state": [2, "allow-in-func"],
+		"react/no-did-update-set-state": "error",
 		// Don't allow the state to be mutated. Ever.
-		"react/no-direct-mutation-state": 2,
+		"react/no-direct-mutation-state": "error",
+		// Don't allow findDOMNode, prefer using a callback ref instead.
+		// https://github.com/yannickcr/eslint-plugin-react/issues/678#issue-165177220
+		"react/no-find-dom-node": "error",
 		// Don't allow "isMounted()" calls.
-		"react/no-is-mounted": 2,
+		"react/no-is-mounted": "error",
 		// DISABLING, THERE ARE TIMES WHEN THIS SHOULD BE ALLOWED
 		// Only allow one component to be defined per file.
-		"react/no-multi-comp": 0,
+		"react/no-multi-comp": "off",
+		// Don't use the returned element from rendering with ReactDOM.
+		"react/no-render-return-value": "error",
 		// Allow local state to be changed, even though we're using Redux.
-		"react/no-set-state": 0,
+		"react/no-set-state": "off",
 		// Don't allow string references.
-		"react/no-string-refs": 2,
+		"react/no-string-refs": "error",
+		// Don't allow (usually accidentally) unescaped entities from appearing.
+		"react/no-unescaped-entities": "error",
 		// Don't allow unknown properties.
-		"react/no-unknown-property": 2,
+		"react/no-unknown-property": "error",
+		// Prevent setState in componentDidUpdate
+		"react/no-will-update-set-state": "error",
 		// Only allow ES6 classes.
-		"react/prefer-es6-class": [2, "always"],
+		"react/prefer-es6-class": ["error", "always"],
+		// Prefer stateless functions whenever possible.
+		"react/prefer-stateless-function": "error",
 		// Require prop types to be defined.
-		"react/prop-types": 2,
+		"react/prop-types": "error",
 		// Prevent React not being defined when using JSX
-		"react/react-in-jsx-scope": 2,
-		// Only allows .js extensions
-		"react/require-extension": [2, {
-			"extensions": [".js"]
-		}],
+		"react/react-in-jsx-scope": "error",
+		// Don't require defaultProps for props that are not required.
+		// We often use optional props to selectively hide elements in a component if the prop is not
+		// provided, this would be detrimental to that practice.
+		"react/require-default-props": "off",
+		// Don't require optimization -- yet.
+		// This would require every compnent to have a shouldComponentUpdate defined.
+		"react/require-optimization": "off",
+		// Don't allow people to forget to return inside render.
+		"react/require-render-return": "error",
 		// Don't allow extra closing tags for components without children.
-		"react/self-closing-comp": 2,
+		"react/self-closing-comp": "error",
```

Comment:
I understand that fear, but I still don't share it. I think another thing I'm not convinced on is that there are that many html tags that are, in fact, self-closing in a logical way in react. The example provided in the documentation seems, really, quite silly. Why would you ever want a container div that is empty?

More generally, though, I'm having trouble thinking of what the use case is for most html tags being self closing. The predominate case I can think of where there are empty divs are, rather, in other approaches to the frontend -- like vanilla JS or jQuery where we target a div and manipulate it.

I'm open to considering this more, but both your example and the example from the documentation seem unrealistic. I'd also add that, to me, part of the benefit of self-closing components (html or react components) is that the signify an intentional finality. An empty div doesn't do that.